### ISSUE TITLE
Adding training / validation / testing sub-datasets within the Observables datasets

### ISSUE THREAD

**Issue by epaillas** [epaillas/acm#98]
We have gotten a request to include information about the training / validation / test set as part of the compressed datasets. If that split is already decided by the time the files are being compressed, we could just add some entries here:

https://github.com/epaillas/acm/blob/a0ba7bf99fcb4f79adb92172c2e8b462c20cb733/acm/observables/emc/spectrum.py#L183

so that apart from 'x' we also have 'x_train', 'x_val', 'x_test', etc, which would be subsets of 'x'. Same thing for 'y'.

@PalPaulD I will try to implement an example for the galaxy power spectrum and then you can tell me if that's sufficient.

**Comment by SBouchard01** [comment]
I can make a quick proof of concept now and take a closer look on Monday :)

**Comment by epaillas** [comment]
(Also taking into account that we might have more HOD samples in the test set tham we do in the training set... this is definitely happening for some statistics already, but I haven't actually implemented that in terms of code because we don't have that functionality yet. Having something like obs.x_test would make it easy I think).

**Comment by SBouchard01** [comment]
As a note, I thought `sunbird` was handling the split between training and validation set? Do we still need to split this one ourselves?

**Comment by SBouchard01** [comment]
PR merged, I'll make a pass on the EMC observables soon to add the train/test split in the compression methods, as specified in #106
